apps/webpack.js

```diff
@@ -248,30 +249,47 @@ function devtool(options) {
   }
 }
 
-var storybookConfig = _.extend({}, baseConfig, {
-  devtool: devtool(),
-  resolve: _.extend({}, baseConfig.resolve, {
-    alias: _.extend({}, baseConfig.resolve.alias, {
-      '@cdo/apps/lib/util/firehose': path.resolve(__dirname, 'test', 'util')
-    })
-  }),
-  externals: {
-    blockly: 'this Blockly'
-  },
-  plugins: [
-    new webpack.ProvidePlugin({React: 'react'}),
-    new webpack.DefinePlugin({
-      IN_UNIT_TEST: JSON.stringify(false),
-      IN_STORYBOOK: JSON.stringify(true),
-      'process.env.mocha_entry': JSON.stringify(process.env.mocha_entry),
-      'process.env.NODE_ENV': JSON.stringify(
-        envConstants.NODE_ENV || 'development'
-      ),
-      PISKEL_DEVELOPMENT_MODE: JSON.stringify(false)
-    }),
-    new webpack.IgnorePlugin(/^serialport$/)
-  ]
-});
+// Customize webpack config for storybook.
+// @param {Object} sbConfig - Webpack configuration from storybook library.
+function storybookConfig(sbConfig) {
```

Comment:
For my understanding, how does this work differently now? Did storybook not provide webpack config previously?

Reply:
i _think_ storybook always provided some webpack configuration and would try to merge theirs with ours (previously provided by `apps/.storybook/webpack.config.js`) and it caused bugs and annoying confusion. so now there is a config function that supplies the storybook config and you can pick-and-choose what you want or just overwrite it (although i think that might break some storybook features if you just ignored their config)

Reply:
For my understanding, how did you decide where to include `sbConfig` when building the returned object? eg, no `...sbConfig.resolve.alias` in `resolve.alias`?

Reply:
mostly guess-and-check 😅 i knew which configurations we cared about from our original `storybookConfig` options, so i tested out which options we needed from storybook and which we could overwrite/exclude without breaking anything